### PR TITLE
allow watch for recursive not existing folder

### PR DESCRIPTION
**Pull request by exx8**
Solves #872 

This pull request allows user to watch specific not existing folders.
Was achieved via introducing a new function _mostSpecifiedExistingAncestor
We should consider making  the function public, as users might be interested in this functionality.

### CODE REVIEW

**Comment by paulmillr** [comment]
We need tests for that feature. Also, ensure you conform to chokidar code style e.g. `{` locations.

**Comment by exx8** [comment]
I added tests.

File: index.js [hunk 1]
```diff
@@ -765,15 +766,37 @@ _isntIgnored(path, stat) {
...
 
+  /**
+   * Find the most recent (specified ) ancestor path that exists.
+   * @param {Path} path file or directory
+   * @returns {Promise<String>} the most recent ancestor path that exists
+   * @private
+   */
+  async _mostSpecifiedExistingAncestor(path)
+{
+  let father=globParent(path);
```
Comment (paulmillr) [inline index.js]: You need to conform to existing code style. That means using constants, proper indentation etc.

**Comment by exx8** [comment]
I've made the changes that you asked.

**Comment by exx8** [comment]
Ping?

**Comment by paulmillr** [comment]
If i’m reading this correctly, we’ll have a huge performance degradation because of the new IO call for every file. Need to investigate this thoroughly and so some benchmarks.

**Comment by exx8** [comment]
> If i’m reading this correctly, we’ll have a huge performance degradation because of the new IO call for every file. Need to investigate this thoroughly and so some benchmarks.

maybe we should use a flag to enable or disable this functionality?

**Comment by XhmikosR** [comment]
This needs a rebase. Also the curly brackets shouldn't be placed on new lines, there's an extra space in a test message and extra newlines.

**Pull request closed without merging**